### DISCUSSION TITLE
googlevideo.com URL Download Speed

### DISCUSSION THREAD

**Discussion by r0hin** [fent/node-ytdl-core#992] [Q&A] [answered]
Hi! I've written up a function to download the audio file from the googlevideo.com URL returned by this library. Here is the code:
```
progress(request(url), {})).on('progress', (state) => {
    console.log(state.percent * 100);
}).on('end', () => {
    console.log('Done')
})
```

And I noticed that progress is very non-linear. 
![image](https://user-images.githubusercontent.com/46874702/129284372-d2ac17e5-1e09-4820-a56d-22195055a0a7.png)
Above is the list of percentages returned by the download function. The packages used are `request` and `request-progress`. Theres a pattern where the number goes up a small amount, then a large amount, then theres are **long** delay before the process repeats. This happens both locally and on a heroku server so it must be an issue with the request itself or a NodeJS option i'm missing.

Downloading the audio file from the URL using a standard browser like chrome works perfectly fine and downloads it entirely within a second. What am I missing?

Thanks for the help!

**Comment by TimeForANinja** [comment, marked as answer]
request is deprecated
so that might already be it

other than that google's chrome might do some trickery like doing parallel ranged requests to speed up the download

but that's just guessing 🤷

**Reply by r0hin** [reply]
Thx for the reply! I did find a solution though. Strangely enough the download speeds increase greatly if I include this as a header: 
``` 
headers: {
    "range": "bytes=0-",
}
```
